### PR TITLE
cleaned up mocker usage

### PR DESCRIPTION
**Pull request by mitchelljkotler**

### CODE REVIEW

File: documentcloud/documents/entity_extraction.py [hunk 1]
```diff
@@ -223,7 +224,7 @@ def _get_mid_info(mids):
     return info
 
 
-def _get_or_create_entities(entities, retries=2, bulk_create=EntityOccurrence.objects.bulk_create):
+def _get_or_create_entities(entities, retries=2):
```
Comment (mitchelljkotler) [inline documentcloud/documents/entity_extraction.py:227]: No need to edit the method directly!

File: documentcloud/documents/tests/test_entity_extraction.py [hunk 2]
```diff
@@ -4,252 +4,78 @@
...
 
 # DocumentCloud
 from documentcloud.core.tests import run_commit_hooks
-from documentcloud.documents.models import Entity
 from documentcloud.documents.entity_extraction import _get_or_create_entities
+from documentcloud.documents.models import Entity
 
 # pylint: disable=too-many-lines, too-many-public-methods
 
 entity_dict = {
```
Comment (mitchelljkotler) [inline documentcloud/documents/tests/test_entity_extraction.py:21]: Auto-formatter made a lot of incidental changes

File: documentcloud/documents/tests/test_entity_extraction.py [hunk 3]
```diff
@@ -4,252 +4,78 @@
...
+        {"text": {"content": "Professor", "begin_offset": 32518}, "type_": 2},
+        {"text": {"content": "Knight", "begin_offset": 43223}, "type_": 1},
+        {"text": {"content": "Professor", "begin_offset": 43213}, "type_": 2},
+    ],
 }
 
+
 @pytest.mark.django_db()
 class TestEntityExtraction:
-    def mock_bulk_create(self, entity_objs):
```
Comment (mitchelljkotler) [inline documentcloud/documents/tests/test_entity_extraction.py:236]: This method is not needed

File: documentcloud/documents/tests/test_entity_extraction.py [hunk 4]
```diff
@@ -4,252 +4,78 @@
...
-        self.entity_objs_passed_to_bulk_create = entity_objs
-        print("entity_objs", entity_objs)
-
-    def test_get_or_create_entities(self):
+    def test_get_or_create_entities(self, mocker):
```
Comment (mitchelljkotler) [inline documentcloud/documents/tests/test_entity_extraction.py:64]: use the mocker fixture

File: documentcloud/documents/tests/test_entity_extraction.py [hunk 5]
```diff
@@ -4,252 +4,78 @@
...
         """get or create entities"""
-        self.entity_objs_passed_to_bulk_create = None
-        #entity = Entity(**entity_dict)
-        #assert entity
-        #pdb.set_trace()
+        mock_bulk_create = mocker.patch(
```
Comment (mitchelljkotler) [inline documentcloud/documents/tests/test_entity_extraction.py:66]: Patch the bulk_create method

**Review by jimkang — APPROVED** [review]
Thanks! The formatting is nice, too.

File: documentcloud/documents/tests/test_entity_extraction.py [hunk 6]
```diff
@@ -4,252 +4,78 @@
...
+
+        _get_or_create_entities([entity_dict])
 
-        # Mock
-        _get_or_create_entities([entity_dict], bulk_create=self.mock_bulk_create)
-        assert self.entity_objs_passed_to_bulk_create, "Entity objects were not passed to bulk_create."
-        assert len(self.entity_objs_passed_to_bulk_create) == 1, "The number of entity objects passed to bulk_create are incorrect."
-        entity_obj = self.entity_objs_passed_to_bulk_create[0]
-        #pdb.set_trace()
+        assert mock_bulk_create.call_args.args[
```
Comment (mitchelljkotler) [inline documentcloud/documents/tests/test_entity_extraction.py:72]: This is in the documentation I sent you in more detail, but the mock object has a property called `call_args` which contains the arguments the mock was last called with - this contains both `args` and `kwargs`.  We want the first positional argument, hence `.call_args.args[0]`

**Review by mitchelljkotler — COMMENTED** [review]
Hope this is informative on how to do what you were trying to do with mocker.  Let me know if you have any other questions and refer to the documentation on mock for other functionality.

**Pull request merged**